### PR TITLE
chore: Update Safe dependencies to their new SDKS

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Safe has a new set of libraries and NPM packages. I updated our codebase so it doesn't depend any more in Gnosis Safe, and uses the new ones. 

## Testing
Test Gnosis Safe App and Wallet Connect Safe and verify that this PR doesn't break anything

Fot the safe app testing, you can add a custom app with URL: https://swap-dev-git-update-safe-cowswap.vercel.app/

### CODE REVIEW

File: src/cow-react/api/gnosisSafe/index.ts [hunk 1]
```diff
@@ -19,35 +20,41 @@ const CHAIN_SHORT_NAME: Partial<Record<number, string>> = {
   [ChainId.GOERLI]: 'gor', // https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-5.json
 }
 
-const SAFE_TRANSACTION_SERVICE_CACHE: Partial<Record<number, SafeServiceClient | null>> = {}
+const SAFE_TRANSACTION_SERVICE_CACHE: Partial<Record<number, SafeApiKit | null>> = {}
 
-function _getClient(chainId: number, library: Web3Provider): SafeServiceClient | null {
+const txServiceUrl = 'https://safe-transaction-goerli.safe.global'
```
Comment (shoom3301) [inline src/cow-react/api/gnosisSafe/index.ts:25]: I guess we should have a map for all networks

Comment (anxolin) [reply src/cow-react/api/gnosisSafe/index.ts]: This was not needed. We already have the URL by network, it was just a test. 

I just removed it and use the right url

**Review by shoom3301 — APPROVED** [review]

**Comment by anxolin** [comment]
> The only one issue I faced is that onchain TXs are not finalized (hang in pending state) when connect to the Safe mobile app

I had to play a bit hide and seek with this one, but I think i found it. It was not detecting it was a Safe because we were checking the name "Gnosis Safe" and not "Safe"

**Comment by anxolin** [comment]
@elena-zh I just checked and it works now as before

image


Maybe in the test, is nice to test the 3 networks. Not urgent tests though

**Review by elena-zh — APPROVED** [review]
Hey @anxolin , great changes!!

Now, when connected to the Safe mobile app, the safe block is back for transactions in the Recent activity! (Bit not for a desktop app when connected to the Safe web app using WC).

While testing the PR, I have found 2 regression issues^
 - reproducible on Develop only: #2392
 - reproducible on Prod: #2393 

But they are not related to your changes and will be addresses in separate PRs. 

Thank you!

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**